test/integration/targets/pids/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+# Test code for the pid module
+# Copyright 2018, Saranya sridharan
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: "Installing PidOf"
+  yum:
+    name: sysvinit-tools
+    state: present
+  when: ansible_distribution == 'CentOS'
+
+  
+- name:  "Installing PidOf in freebsd"  
+  pkgng:
+    name: pidof
+    state: present
+  when: ansible_distribution == 'FreeBSD'
+  
+- name: Install  PidOf in (OSX)
+  block:
+    - name: Find brew binary
+      command: which brew
+      register: brew_which
+    - name: Get owner of brew binary
+      stat: path="{{ brew_which.stdout }}"
+      register: brew_stat
+    - name: "Install package"
+      homebrew:
+        name: pidof
+        state: present
+        update_homebrew: no
+      become: yes
+      become_user: "{{ brew_stat.stat.pw_name }}"
+      
+  when: ansible_os_family == 'Darwin'
+
+- name: "Installing setproctitle"
+  pip:
+    name: setproctitle
+    version: 1.1.10
+
+- name: "Checking empty result" 
+  pids: 
+    name: "blahblah"
+  register: emptypids
+  
+   
```

Comment:
Here (and below): please use only one newline between tasks, if there isn't a good reason for having more. In this place at least, it doesn't make sense that the above task is separated by its predecessors by only one newline and then the next task (which belongs to the above task) is suddenly separated by two newlines.